code/modules/projectiles/gun.dm

```diff
@@ -327,6 +327,8 @@
 
 
 
+	if(istype(target, /obj/structure/catwalk) || istype(target, /obj/item))
```

Comment:
Putting specific typechecks into unrelated code like this is bad. This generates unmaintainable spaghetti code

Find a better approach

Reply:
Done